certbot/tests/main_test.py

```diff
@@ -1012,12 +1014,12 @@ def test_certonly_csr_dry_run(self):
     @mock.patch('certbot.main.client.acme_client')
     def test_revoke_with_key(self, mock_acme_client):
```

Comment:
I'd add another test below this one named something like: `test_revoke_with_mismatched_key` and make sure the appropriate error is raised to make sure revoking by key is failing when it's supposed to. You can use the previous values of `CERT` and `KEY` to make it fail.